[afiqzu] iP

afiqzu opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,

- text based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is __FREE__!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

afiqzu committed:
added greet, echo, exit

afiqzu committed:
added add and list functionality

afiqzu committed:
added Task class and 'mark as done' functionality

afiqzu committed:
added Deadline, Event, and ToDo classes

afiqzu committed:
fixed line formatting and added input commands for testing

afiqzu committed:
Added DukeException and fixed minor typos

afiqzu committed:
no message

afiqzu committed:
Revert "no message"

This reverts commit 3385f2e8c16ba5cd20ed37afda53214fd23ef99e.

afiqzu committed:
added delete functionality

afiqzu committed:
created new input testcases

afiqzu committed:
Add save functionality

afiqzu committed:
Teach Duke to understand date and time

afiqzu committed:
Merge branch 'branch-Level-8'

afiqzu committed:
Refactor the code to extract out closely related code as classes

afiqzu committed:
Divide class into duke package

afiqzu committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

afiqzu committed:
Add JUnit tests

afiqzu committed:
Configure gradle to create JAR file

afiqzu committed:
Add documentation

afiqzu committed:
Tweak code to comply with coding standard

afiqzu committed:
Add 'find' functionality

afiqzu committed:
Merge branch 'branch-A-CodingStandard' and resolve conflicts

afiqzu committed:
Merge branch 'branch-Level-9' and resolved conflicts

afiqzu committed:
Create command class

afiqzu committed:
Refactor command class into command subclasses

afiqzu committed:
Add exception and task package

zenithyap commented on src/main/java/duke/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,16 @@
+package duke.command;
+
+import duke.*;
```
Could list out the imports explicitly instead.

zenithyap commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,51 @@
+    @Override
+    public String toLog() {
+        return "E" + super.toLog() + " | "
+                + this.startTime.format(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"))
```
DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm") could be stored as variable in the superclass as it is used quite often.

zenithyap commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,19 @@
+
+import duke.*;
+
+public abstract class Command {
```
Could you add javadocs here?

zenithyap commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,157 @@
+package duke.command;
```
Should the Parser class be in the command package?

zenithyap commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,157 @@
+     * @return String representation of keyword
+     */
+    public static String getFindKeyword(String input) {
+        return input.substring(5);
```
Maybe you could store the 5 as a constant variable instead.

afiqzu committed:
Update README.md

afiqzu committed:
Edit code to comply with checkstyle

afiqzu committed:
Add GUI

afiqzu committed:
Add more GUI code

afiqzu committed:
Customise GUI

bryanongjx commented on src/main/java/duke/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,23 @@
+package duke.command;
```
missing docs for the class! :(

bryanongjx commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,156 @@
+            return new UnmarkCommand(userInput);
+        case "find":
+            return new FindCommand(userInput);
+        case "todo":
```
nice that you didnt repeat return new TaskCommand for all three cases!

bryanongjx commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,22 @@
+
+    @Override
+    public String execute(TaskList tasks) {
+        int toUnMark = userInput.charAt(7) - 48;
```
48 seems abit hardcoded

bryanongjx commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+     * Constructor for Task
+     * @param name of task
+     */
+    public Task(String name) {
```
nice use of 2 different constructors

afiqzu committed:
Add assert feature

Some methods in Parser class require a valid input to work.

As such, the assert function is used at the beginning of these methods
to ensure that the input is valid so that the method body can run
correctly.

afiqzu committed:
Delete minor redundant code

afiqzu committed:
Improve code quality

Methods found in Parser and command classes are too long.

Let's refactor some lines of code from these methods as well as create helper
methods and new command classes for Todo, Deadline, and Event tasks.

afiqzu committed:
Merge pull request #2 from afiqzu/branch-A-Assertions

branch-A-assertions

afiqzu committed:
Merge branch 'master' into branch-A-CodeQuality and resolve conflicts

# Conflicts:
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/AddTaskCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/FindCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/parser/Parser.java

afiqzu committed:
Merge pull request #3 from afiqzu/branch-A-CodeQuality

Improve code quality

afiqzu committed:
Support managing notes

Previous iteration did not support managing notes.

Let's create new command classes such as "NoteCommand"
and "OpenNoteCommand" to support new functionalities such
as adding a new note, as well as viewing current notes.

afiqzu committed:
Improve GUI

afiqzu committed:
Fix bugs

Task Log and Notes did not save to their respective files properly.

Let's fix above mentioned problems.

afiqzu committed:
Add prompt text in input box

afiqzu committed:
Edit GUI

afiqzu committed:
Rename chatbot to 'Jeno'

Chatbot's name was previously set as the default 'Duke' name.

Let's refactor all lines of code and change all instances
of 'duke' to 'jeno'.

afiqzu committed:
Improve code to handle more errors

afiqzu committed:
Add user interface image

afiqzu committed:
Add line spacing for JenoDialogBox

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Fix minor bugs

afiqzu committed:
Add 'help' functionality

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Update README.md

afiqzu committed:
Add minor changes to GUI

afiqzu committed:
Fix checkstyle rule violations
